shared/middleware/files_api.rb

```diff
@@ -477,9 +477,15 @@ def files_put_file(encrypted_channel_id, filename, body)
       manifest = JSON.load manifest_result[:body]
     end
 
-    # store the new file
+    # store the new file (possibly copied from an existing file)
     if params['src']
-      new_entry_json = copy_file('files', encrypted_channel_id, filename.downcase, params['src'])
+      if filename.downcase == params['src']
+        # The rename is a case-only rename. As the filename is downcased on s3, there is no need to
+        # update s3.
+        new_entry_json = manifest.detect {|e| e['filename'].downcase == filename.downcase}.to_json
```

Comment:
There's some repeated work here with the `manifest.detect` operation about ten lines below, where we do a lookup in the manifest again to check whether the hash changed.  It's possible catching this case here is the most elegant solution, but I wonder if we can construct a cleaner path for handling manifest-only updates where we only need to look up the existing manifest entry once and we _know_ the manifest has changed.

Worth writing tests before attempting such a refactor, though.